Fix tests

- Trying to exec /dev/null/calc.exe on FreeBSD will fail with ENOENT
  (xe exit 127) but on Linux fails with ENODIR (xe exit 126)
- The argscap check returns 8186 + 8186 + 1339 = 17711 on FreeBSD,
  which should be fine.

--- tests.orig	2017-10-30 15:43:40 UTC
+++ tests
@@ -220,7 +220,7 @@ check_output 'exit code when process was killed' 'nech
 EOF
 
 # possible false positive result when exec returns ENOENT instead of ENOTDIR here
-check_output 'exit code when command cannot be run' 'necho a | xe /dev/null/calc.exe 2>/dev/null || echo $?' <<EOF
+check_output 'exit code when command cannot be run' 'necho a | xe /dev/null 2>/dev/null || echo $?' <<EOF
 126
 EOF
 
@@ -266,10 +266,8 @@ EOF
 
 printf '# limit checks, expecting maximal POSIX limits available\n'
 
-check_output 'argscap check' 'dd if=/dev/zero bs=1 count=17711 2>/dev/null | tr "\0" "\012" | xe -N0 -s "echo \$#"' <<EOF
-8187
-8187
-1337
+check_output 'argscap check' 'dd if=/dev/zero bs=1 count=17711 2>/dev/null | tr "\0" "\012" | xe -N0 -s "echo \$#" | awk "{sum+=\$0}END{print sum}"' <<EOF
+17711
 EOF
 
 bloat() { perl -e 'print "x"x8000, "\n" for 1..42'; }
